lsp/io.ml

```diff
@@ -74,9 +74,11 @@ let read_request ic =
   | Invalid_argument msg -> raise (ReadError msg)
 
 let mut = Mutex.create ()
+let log = ref (fun _ _ -> ())
```

Comment:
I'm not too fond of this hack: can you try `let ... and let` instead?

Reply:
Indeed that's a good idea, but I have another PR to improve some of the limitations of `trace` logging where this goes away.